chkstat.c

```diff
@@ -492,19 +598,21 @@ safe_open(char *path, struct stat *stb, uid_t target_uid, bool *traversed_insecu
 
       /* owner of directories must be trusted for setuid/setgid/capabilities as we have no way to verify file contents */
       /* for euid != 0 it is also ok if the owner is euid */
-      if (stb->st_uid && stb->st_uid != euid && p)
+      if (stb->st_uid && stb->st_uid != euid && !is_final_path_element)
         *traversed_insecure = true;
       // path is in a world-writable directory, or file is world-writable itself.
-      if (!S_ISLNK(stb->st_mode) && (stb->st_mode & S_IWOTH) && p)
+      if (!S_ISLNK(stb->st_mode) && (stb->st_mode & S_IWOTH) && !is_final_path_element)
         *traversed_insecure = true;
       // if parent directory is not owned by root, the file owner must match the owner of parent
       if (stb->st_uid && stb->st_uid != target_uid && stb->st_uid != euid)
         {
-          if (p)
-            goto fail_insecure_path;
+          if (is_final_path_element)
+            {
+              //fprintf(stderr, "%s: has unexpected owner. refusing to correct due to unknown integrity.\n", path+rootl);
```

Comment:
Could you also add the reasoning as in SLE-15-SP1 here? There you have:

`// do not backport behavior change`

Reply:
Done. That's a remnant of manually editing patch files in different code streams. (I think one also uses "don't" instead of "do not".)

Reply:
I think there's no strict rule to use "don't", actually I've been told in formal texts "do not" is to be preferred.